engine/apps/alerts/tasks/check_escalation_finished.py

```diff
@@ -115,6 +115,55 @@ def check_personal_notifications_task(alert_group_id) -> None:
         task_logger.info(f"{base_msg} personal notifications check passed")
 
 
+@shared_task
+def check_personal_notifications_task() -> None:
+    """
+    This task checks that triggered personal notifications are completed.
+    It will log the triggered/completed values to be used as metrics.
+
+    Attention: don't retry this task, the idea is to be alerted of failures
+    """
+    from apps.alerts.models import AlertGroup
+    from apps.base.models import UserNotificationPolicy, UserNotificationPolicyLogRecord
+
+    # use readonly database if available
+    readonly_db = get_random_readonly_database_key_if_present_otherwise_default()
+
+    now = timezone.now()
+
+    # consider alert groups from the last 2 days
+    alert_groups = AlertGroup.objects.using(readonly_db).filter(
+        started_at__range=(now - timezone.timedelta(days=2), now),
+    )
+
+    # review notifications triggered in the last 20-minute window
+    # (task should run periodically about every 15 minutes)
+    since = now - timezone.timedelta(minutes=20)
+
+    log_records_qs = UserNotificationPolicyLogRecord.objects.using(readonly_db)
+    # personal notifications triggered in the given window for those alert groups
+    triggered = log_records_qs.filter(
+        type=UserNotificationPolicyLogRecord.TYPE_PERSONAL_NOTIFICATION_TRIGGERED,
+        notification_step=UserNotificationPolicy.Step.NOTIFY,
+        created_at__gte=since,
+        created_at__lte=now,
+        alert_group__in=alert_groups,
+    ).count()
+
+    # personal notifications completed in the given window for those alert groups
+    completed = log_records_qs.filter(
+        Q(type=UserNotificationPolicyLogRecord.TYPE_PERSONAL_NOTIFICATION_FAILED)
+        | Q(type=UserNotificationPolicyLogRecord.TYPE_PERSONAL_NOTIFICATION_SUCCESS),
+        notification_step=UserNotificationPolicy.Step.NOTIFY,
+        created_at__gt=since,
+        created_at__lte=now,
+        alert_group__in=alert_groups,
+    ).count()
+
+    task_logger.info(f"Personal notifications triggered: {triggered}")
+    task_logger.info(f"Personal notifications completed: {completed}")
```

Comment:
Would these be better as a single log line? If we are querying logs over wider time ranges to produce metrics it will be easier to have them paired rather than using the task id to correlate.  Also `notifications_triggered={triggered} notifications_completed={completed}` has nicer natural behavior with logfmt in loki.